test/sqllogictest/joins.slt

```diff
@@ -68,3 +68,56 @@ query ITIT colnames,rowsort
 SELECT * FROM l JOIN r ON l.la = r.ra LIMIT 0
 ----
 la  lb  ra  rb
+
+# Test that projections work through join plans with scalars.
+query ITT rowsort
+SELECT l1.la, l2.lb, l3.lb
+FROM l as l1, l as l2, l as l3
+WHERE l1.la + 1 = l2.la AND l3.la = l1.la + l2.la
+----
+1  l2  l3
+
+# Test that join plans with scalars work in subqueries
```

Comment:
Just add an explain plan test for this.

Reply:
The plan will 100% be something we *don't* want to maintain, is why I didn't include it.

Reply:
I'm guessing we want to improve the query planning significantly for this one? Alternatively, we can add a "TODO: explain plan for this when better planning around subqueries comes about"

Reply:
I have a note that it is totally fine to change the plan. I don't think this particular one is the right place to track "subqueries are badly planned". I think the TODO isn't necessary in that the plan should break if better planning comes around. Lmk if that doesn't sound right to you!

Reply:
:+1: